fastplotlib/layouts/_graphic_methods_mixin.py

```diff
@@ -432,6 +433,208 @@ def add_line_stack(
+            These can e.g. be some domain-specific value, mapped to [0..1].
+            If ``mapcoords`` and ``cmap`` are given, they are used instead of ``colors``.
+
+        cmap: str, optional
+            Apply a colormap to the mesh, this overrides any argument passed to
+            "colors". For supported colormaps see the ``cmap`` library
+            catalogue: https://cmap-docs.readthedocs.io/en/stable/catalog/
+            Both 1D and 2D colormaps are supported, though the mapcoords has to match the dimensionality.
+
+        **kwargs
+            passed to :class:`.Graphic`
+
+
+        """
+        return self._create_graphic(
+            MeshGraphic,
+            positions,
+            indices,
+            colors,
+            mapcoords,
+            cmap,
+            isolated_buffer,
+            **kwargs,
+        )
+
+    def add_surface(
+        self,
+        data: Any,
+        colors: Union[str, numpy.ndarray, Sequence] = "w",
+        mapcoords: Any = None,
+        cmap: str = None,
+        clim: tuple[float, float] | None = None,
+        isolated_buffer: bool = True,
+        **kwargs,
+    ) -> MeshGraphic:
+        """
+
+        Create a mesh Graphic
+
+        Parameters
+        ----------
+        data: array-like
+            A height-map (an image where the values indicate height, i.e. z values).
+            Can also be a 3-tuple to explicitly specify the x and y values in addition to the z values.
+
+        colors: str, array, or iterable, default "w"
+            A uniform color, or the per-position colors.
+
+        mapcoords: array-like
+            The per-position coordinates to which to apply the colormap (a.k.a. texcoords).
+            These can e.g. be some domain-specific value (mapped to [0..1] using ``clim``).
+            If not given, they will be the depth (z-coordinate) of the surface.
+
+        cmap: str, optional
+            Apply a colormap to the mesh, this overrides any argument passed to
+            "colors". For supported colormaps see the ``cmap`` library
+            catalogue: https://cmap-docs.readthedocs.io/en/stable/catalog/
+            Both 1D and 2D colormaps are supported, though the mapcoords has to match the dimensionality.
+
+        clim: tuple[float, float]
+            The colormap limits. If the mapcoords has values between e.g. 5 and 90, you want to set the clim
+            to e.g. (5, 90) or (0, 100) to determine how the values map onto the colormap.
+
+        **kwargs
+             passed to :class:`.Graphic`
+
+
+        """
+
+        def check_z(z):
+            if z.ndim != 2:
+                raise ValueError("Z must be a 2D array.")
+
+        # In VisVis (https://github.com/almarklein/visvis/blob/main/visvis/functions/surf.py)
+        # the tuple can be 2-element or 4-element, to pass a color per z-value.
+        # I disabled that by commenting related logic. Maybe it can be enabled someday.
```

Comment:
Been playing around with this, I think it would be clearer if there's an explicit kwarg for this, can look into it later when a usecase arises.